.github/workflows/tpu-testing.yml

```
@@ -0,0 +1,96 @@
+name: Run TPU tests on current PR.
+
+on:
+  pull_request:
+    branches:
+    - master
+
+env:
+  PROJECT_ID: ${{ secrets.GKE_PROJECT }}
+  GKE_CLUSTER: oneshots-europe-west4   # TODO: update to cluster name.
+  GKE_ZONE: europe-west4-a  # TODO: update to cluster zone.
+  GKE_REGION: europe-west4  # TODO: update to cluster region.
+  IMAGE: gcr.io/${{ secrets.GKE_PROJECT }}/tpu-testing-image
+
+jobs:
+  setup-build-publish-deploy:
+    name: tpu-testing-job
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14.x
+
+    - name: Checkout Pytorch Lightning
+      uses: actions/checkout@v2
+      with:
+        repository: PyTorchLightning/pytorch-lightning
+        ref: ${{ github.event.pull_request.head.sha }}
+        path: main
+
+    - name: Checkout ml-testing-accelerators
+      uses: actions/checkout@v2
+      with:
+        repository: GoogleCloudPlatform/ml-testing-accelerators
+        path: ml-testing-accelerators
+        ref: 4ebcf060a110a359c2f214d81be7334313ce3c03
+
+    # Setup gcloud CLI
+    - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+      with:
+        version: '290.0.1'
+        service_account_key: ${{ secrets.GKE_SA_KEY_BASE64 }}
+        project_id: ${{ secrets.GKE_PROJECT }}
+        export_default_credentials: true
+
+    # Configure Docker to use the gcloud command-line tool as a credential
+    # helper for authentication
+    - run: |-
+        gcloud --quiet auth configure-docker
+
+    - name: Build and Push Docker Image
+      run: |-
+        cd main/docker/tpu && \
+        docker build --tag "$IMAGE:$GITHUB_RUN_ID" -f Dockerfile --build-arg "GITHUB_REF=$GITHUB_REF" --build-arg "TEST_IMAGE=1" . && \
+        docker push "$IMAGE:$GITHUB_RUN_ID"
+
+    # Install jsonnet.
+    - run: |-
+        go get github.com/google/go-jsonnet/cmd/jsonnet
+
+    # Get the GKE credentials so we can deploy to the cluster
+    # Use either zone or region depending on cluster setup.
+    - run: |-
+        #gcloud container clusters get-credentials "$GKE_CLUSTER" --zone "$GKE_ZONE"
+        gcloud container clusters get-credentials "$GKE_CLUSTER" --region "$GKE_REGION"
+
+    - run: |-
```

just curious, do we need everywhere `&& \` which makes the reading a bit floating
```suggestion
    - run: |
```